lib/scripts/windows/redirect_logs.ps1

```diff
@@ -0,0 +1,44 @@
+$hostname = "localhost"
+$port = $Env:STD_LOG_COLLECTION_PORT # 10514
+
+# Recreate the socket, writer objects in the case that the connection was closed
+# https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient?view=netframework-4.8
+function Recreate-Socket {
+    Write-Host "Connecting to $hostname $port"
+    try {
+        $Socket = New-Object System.Net.Sockets.TCPClient($hostname,$port)
+        $Stream = $Socket.GetStream()
+        $Writer = New-Object System.IO.StreamWriter($Stream)
+        $Writer.AutoFlush = $true;
+        return $Socket, $Writer
+    } catch {
+        Write-Host "Couldn't connect to $hostname $port"
+    }
+}
+
+function Run {
+    $Socket, $Writer = Recreate-Socket
+    while($true)
+    {
+        # Reconnect if the socket is disconnected
+        # (The Datadog Agent times out the connection if it does not receive
+        # logs during 1 minute)
+        If ($Socket.Connected -eq $false) {
```

Comment:
Have you tested the scenario when the agent closes the connection, and made sure we don't lose one line ?

Reply:
Not explicitly, no. Good call. I'll do some testing for that by:
* Starting the buildpack, wait for logs to start coming in
* Wait a few minutes to ensure the connection would be closed (or at least one reset happened)
* Hit the route to trigger specific log lines and confirm they come in
(Would that cover this scenario?)

Reply:
Ok, as we discussed offline, I did some manual testing around this scenario It looks like there was indeed an issue with losing the first log line sent after the TCP connection was closed. I've added some logic to handle that case and it looks like its working. Thanks for catching this